google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -0,0 +1,170 @@
+          AppEngineFlexibleFacet.getFacetType();
+      AppEngineFlexibleFacet facet = FacetManager.getInstance(module).addFacet(
+          facetType, facetType.getPresentableName(), null /* underlyingFacet */);
+
+      // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
+      VirtualFile[] contentRoots = rootModel.getContentRoots();
+      AppEngineProjectService appEngineProjectService = AppEngineProjectService.getInstance();
+      if (contentRoots.length > 0) {
+        facet.getConfiguration().setYamlPath(
+            appEngineProjectService.getDefaultAppYamlPath(contentRoots[0].getPath()));
+        facet.getConfiguration().setDockerfilePath(
+            appEngineProjectService.getDefaultDockerfilePath(contentRoots[0].getPath()));
+      }
+
+      // Only adds deployment run configuration for now. Stackdriver debugger to follow.
+      setupDeploymentRunConfiguration(module, facet);
+
+      CloudSdkService sdkService = CloudSdkService.getInstance();
+      if (!sdkService.validateCloudSdk(cloudSdkPanel.getCloudSdkDirectoryText())
+          .contains(CloudSdkValidationResult.MALFORMED_PATH)) {
+        sdkService.setSdkHomePath(cloudSdkPanel.getCloudSdkDirectoryText());
+      }
+    }
+
+    private void setupDeploymentRunConfiguration(Module module, AppEngineFlexibleFacet facet) {
+      RunManager runManager = RunManager.getInstance(module.getProject());
+      AppEngineCloudType serverType =
+          ServerType.EP_NAME.findExtension(AppEngineCloudType.class);
+      DeployToServerConfigurationType configurationType
+          = DeployToServerConfigurationTypesRegistrar.getDeployConfigurationType(serverType);
+
+      RunnerAndConfigurationSettings settings = runManager.createRunConfiguration(
+          configurationType.getDisplayName(), configurationType.getFactory());
+
+      // Sets the GAE Flex server, if any exists, in the run config.
+      DeployToServerRunConfiguration<?, AppEngineDeploymentConfiguration> runConfiguration =
+          (DeployToServerRunConfiguration<?, AppEngineDeploymentConfiguration>)
+              settings.getConfiguration();
+      RemoteServer<AppEngineServerConfiguration> server =
+          ContainerUtil.getFirstItem(RemoteServersManager.getInstance().getServers(serverType));
+      if (server != null) {
+        runConfiguration.setServerName(server.getName());
+      }
+
+      // Copies the specified app.yaml and Dockerfile paths to the deployment run config.
+      AppEngineDeploymentConfiguration deployConfiguration =
+          new AppEngineDeploymentConfiguration();
+      deployConfiguration.setYamlPath(facet.getConfiguration().getYamlPath());
+      deployConfiguration.setDockerFilePath(facet.getConfiguration().getDockerfilePath());
+
+      // Set logged in user.
+      CredentialedUser user = Services.getLoginService().getActiveUser();
+      if (user != null) {
+        deployConfiguration.setGoogleUsername(user.getEmail());
```

Comment:
Fair enough. The idea was that the run configuration was executable out-of-the-box. But, ever since we started requiring users to select a deployable (to enable the before launch task), they will also need to explicitly select a project, so this code is actually rather defunct.